concent_api/core/payments/bankster.py

```diff
@@ -71,96 +81,168 @@ def claim_deposit(
-    available_provider_claim = (
-        min(
-            service.get_deposit_value(client_eth_address=provider_ethereum_address),  # pylint: disable=no-value-for-parameter
-            settings.ADDITIONAL_VERIFICATION_COST
-        )
-        if concent_use_case == ConcentUseCase.ADDITIONAL_VERIFICATION
-        else 0
+    # Bankster asks SCI about the amount of funds available on the deposit account listed in the DepositClaim.
+    available_funds = service.get_deposit_value(  # pylint: disable=no-value-for-parameter
+        client_eth_address=deposit_claim.payer_deposit_account.ethereum_address
     )
 
-    current_time = get_current_utc_timestamp()
-
-    # Handle requestor claim payment info
-    if available_requestor_claim == 0:
-        requestors_claim_payment_info = ClaimPaymentInfo(0, subtask_cost)
-    else:
-        transaction_hash = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
-            requestor_eth_address=requestor_ethereum_address,
-            provider_eth_address=provider_ethereum_address,
-            value=available_requestor_claim,
-            subtask_id=subtask_id,
-        )
-        requestors_claim_payment_info = ClaimPaymentInfo(
-            amount_paid=available_requestor_claim,
-            amount_pending=subtask_cost - available_requestor_claim,
-            tx_hash=transaction_hash,
-            payment_ts=current_time,
+    # Bankster begins a database transaction and puts a database lock on the DepositAccount object.
+    with transaction.atomic(using='control'):
+        DepositAccount.objects.select_for_update().get(
+            pk=deposit_claim.payer_deposit_account_id
         )
 
-    # Handle provider claim payment info
-    if settings.ADDITIONAL_VERIFICATION_COST == 0 or concent_use_case != ConcentUseCase.ADDITIONAL_VERIFICATION:
-        providers_claim_payment_info = ClaimPaymentInfo(0, 0)
-    elif available_provider_claim == 0:
-        providers_claim_payment_info = ClaimPaymentInfo(0, settings.ADDITIONAL_VERIFICATION_COST)
-    else:
-        transaction_hash = service.cover_additional_verification_cost(  # pylint: disable=no-value-for-parameter
-            provider_eth_address=provider_ethereum_address,
-            value=available_provider_claim,
-            subtask_id=subtask_id,
-        )
-        providers_claim_payment_info = ClaimPaymentInfo(
-            amount_paid=available_provider_claim,
-            amount_pending=settings.ADDITIONAL_VERIFICATION_COST - available_provider_claim,
-            tx_hash=transaction_hash,
-            payment_ts=current_time,
+        # Bankster sums the amounts of all existing DepositClaims that have the same payer as the one being processed.
+        aggregated_requestor_claims = DepositClaim.objects.filter(
+            payer_deposit_account=deposit_claim.payer_deposit_account
```

Comment:
How do you know requestor is the payer here?

Reply:
:+1: done